server/src/assets/assets.service.ts

```diff
@@ -22,19 +22,30 @@ export class AssetsService {
   }
 
   async getAssets(getAssetsDto: GetAssetsDto): Promise<Asset[]> {
-    const { limit, offset, title, ...rest } = getAssetsDto;
-    const whereParams = title ? { title: Like(`%${title}%`), ...rest } : { ...rest };
-
-    return (
-      this.assetsRepository.find({
-        where: whereParams,
-        order: {
-          datePosted: 'DESC',
-        },
-        skip: offset,
-        take: limit,
-      }) || []
-    );
+    const { limit, offset, search, ...rest } = getAssetsDto;
+    if (search) {
+      return this.assetsRepository
+        .createQueryBuilder()
+        .select('asset')
+        .from(Asset, 'asset')
+        .where(`"asset"."searchtitle" @@ plainto_tsquery('english', :query)`, {
+          query: `${search}`,
```

Comment:
Do you know if Typeorm sanitizes this `query` property? if not we should add at least a basic sanitation function with `toLower` `trim` and replace any instances of `drop` with blank text.